docs/firmware/device-sdk/light-db/guide-light-db-observe.md

```diff
@@ -0,0 +1,14 @@
+---
+title: LightDB 'Observe' Application Walkthrough
+sidebar_position: 3
+---
+
+The [observe](https://github.com/golioth/zephyr-sdk/tree/main/samples/lightdb/observe) sample demonstrates another method of retrieving resources using CoAP. The [observe](https://github.com/golioth/zephyr-sdk/tree/main/samples/lightdb/observe) sample is somewhat of an asynchronous version of the [get](https://github.com/golioth/zephyr-sdk/tree/main/samples/lightdb/get) sample. The [observe](https://github.com/golioth/zephyr-sdk/tree/main/samples/lightdb/observe) sample methods are used to establish a subscription to resources in the Golioth cloud. Changes to the state of these resources in the cloud will trigger an update to the known state of the resource on the device.
+
```

Comment:
I'm going to make an issue to add code snippets inline with the guides implemented by this PR.

The issue will not block merging of #106